client/client.go

```diff
@@ -0,0 +1,766 @@
+package client
+
+import (
```

Comment:
@coderabbitai read client/client.go and generate usage examples in Markdown format. Include examples for doing GET, POST, PUT, using TLS config, using JSON for request and response, etc.